lib/src/client/http2_connection.dart

```diff
@@ -61,6 +61,7 @@ class Http2ClientConnection implements connection.ClientConnection {
 
   final String host;
```

Comment:
This is a duplicated code between client and the server around managing `host`, `port` and `unix` flag. I suggest factoring it out into a `SocketAddress` class, which can also handle parsing parsing unix domain sockets addresses. 

For unix domain sockets I suggest using `http+unix` as a scheme which seems to be a canonical for HTTP over UDS. I also suggest using a properly encoded Uri in that case, so that parsing becomes something along the lines of:

```dart
class SocketAddress {
  final Object host; // Either InternetAddress or String.
  final int port;

  SocketAddress.unix(String path)
      : host = InternetAddress(path, type: InternetAddressType.unix),
        port = 0;

  SocketAddress.tcp(String host, this.port)
      : host = host;

  factory SocketAddress(String host, int port) {
    if (host.startsWith('http+unix://')) {
      final path = Uri.parse(host).replace(scheme: 'file').toFilePath();
      return SocketAddress.unix(path);
    } else {
      return SocketAddress.tcp(host, port);
    }
  }
}
```

Reply:
> final path = Uri.parse(host).replace(scheme: 'file').toFilePath();
It does not work:
```
Unhandled exception:
Unsupported operation: Cannot extract a non-Windows file path from a file URI with an authority
#0      _Uri._toFilePath (dart:core/uri.dart:2725:7)
#1      _Uri.toFilePath (dart:core/uri.dart:2720:65)
```

Reply:
Ah, this works only for absolute uris. 

At which point I am thinking that maybe simpler way is just do this:

```
class ClientChannel extends ClientChannelBase {
  /// Host can be either [String] hostname or [InternetAddress] including
  /// a unix domain socket address.
  final Object host;
  final int port;
  final ChannelOptions options;

  ClientChannel(this.host,
      {this.port = 443, this.options = const ChannelOptions()})
      : super();

  @override
  ClientConnection createConnection() {
    return Http2ClientConnection(host, port, options);
  }
}
```

`Server.serve` already can take `InternetAddress` so it should already work.